tools/packaging/common/istio-iptables.sh

```diff
@@ -59,29 +59,8 @@ function dump {
     ip6tables-save
 }
 
-#
-#  Function check if argument is a valid ipv4 or ipv6 address
-#
-function isValidIP() {
-   if [ "$1" != "${1#*[0-9].[0-9].[0-9].[0-9]}" ]; then
-      true
-   elif [ "$1" != "${1#*:[0-9a-fA-F]}" ]; then
-      true
-   else
-      false
-   fi
-}
-
-#
-# Function return true if agrument is a valid ipv4 address
-#
-function isIPv4() {
-   if [ "$1" != "${1#*[0-9].[0-9].[0-9].[0-9]}" ]; then
-      true
-   else
-      false
-  fi
-}
+# Load ip addresses validation functions
+. ./tools/packaging/common/validations.sh
```

Comment:
can you ensure the debian/rpm/sidecar docker include validations.sh script?

Reply:
I could not find a place where `istio-iptables.sh` get explicitely added to the image, so I suspect whole `tools/packaging/common` folder is copied for every image flavour. If you think it is not the case please point me where copying of the file happens.

Reply:
code search on istio-iptables.sh shows multiple references.
https://github.com/istio/istio/search?q=istio-iptables.sh&unscoped_q=istio-iptables.sh

This includes init container, Debian package and rpm package. I don't know all the build details, but I think at least `validations.sh` must be able to be referenced by `istio-iptables.sh` in all those environments.

Also you reference validations.sh as `. ./tools/packaging/common/validations.sh`. This will work with Git workdir, but I doubt istio-iptable.sh will be invoked and has also such reletive path in environments I mentioned above.

I'm not sure how the presubmit test using init container docker image can pass if validation.sh can't be resolved. might not be a problem thought, just not clear to me.

Figuring this out might be a bit of complicated... So thats why i initially suggested to add another command level option in this script just for test. look at `usage()`.

add @jwendell @sdake who might know more.

Reply:
By the way, splitting script to different files is definitely great practice, IMO, just some additional use cases & build stuff need to verify.

Reply:
@incfly If validations.sh were missing in any image it would have never passed CI, as I think docker images are build for e2e tests.

Reply:
There is no good reason to add complexity or have validation as a separate file. 
I'm already concerned that the validation script is too complicated already - and the fact it broke stuff proves it. 
We get the IP from a trusted source - it doesn't need validation, and there are far simpler way to detect if it is ipv6 or ipv4. Just check if it has a ":" in it, or doesn't have "."  

Over complex code leads to bugs, validating the output of ip commands is not needed.

Reply:
maybe in this specific case we get it from the trusted source, but people tend to re-use already developed functions, so there is a good chance it would be reused and with proposed approach it will not validate. Besides this version of validation comes with unit testing which is covering pretty much any variations, so the chances of introducing a bug aI would say fairly low. If you do not like complexity, I am afraid then whole Istio code base needs to be revisited as it is far from being simple.